### ISSUE TITLE
[DependencyInjection] Service names containing underscores not working after upgrading to 6.4.0

### ISSUE THREAD

**Issue by paullallier** [symfony/symfony#52815]
### Symfony version(s) affected

6.4.0

### Description

Sorry - I'm not sure if this is a bug or just a misconfiguration by me.

Upgrading from 6.3.x to 6.4.0, and autowiring Redis is suddenly throwing an error.  

### How to reproduce

I have this, which works in 6.3.9:

```
<?php

declare(strict_types=1);

namespace App\MessageHandler;

use App\Message\ClearUserAccessTokens;
use Redis;
use Symfony\Component\Messenger\Attribute\AsMessageHandler;

#[AsMessageHandler]
readonly class ClearUserAccessTokensHandler
{
    public function __construct(protected Redis $redis_valid_token)
    {
    }
```

When I upgrade to 6.4.0, I get this error:
```
!!  In DefinitionErrorExceptionPass.php line 51:
!!                                                                                 
!!    Cannot autowire service "App\MessageHandler\ClearUserAccessTokensHandler":   
!!    argument "$redis_valid_token" of method "__construct()" references class "R  
!!    edis" but no such service exists. Did you mean to target one of "redis_vali  
!!    d_token", "redis_refresh" instead?                          
```                 

Might this be a bug, or just a config issue on my project (nothing changes apart from upgrading Symfony and a few other packages not related to Redis)

In case it's relevant, I'm using snc/redis-bundle to manage the services, with phpredis:
```
services:
    Redis $redis_valid_token: '@snc_redis.valid_token'
    Redis $redis_refresh: '@snc_redis.refresh'


snc_redis:
    clients:
        refresh:
            type: phpredis
            alias: refresh
            dsn: "%env(REDIS_URL_REFRESH)%"
        valid_token:
            type: phpredis
            alias: valid_token
            dsn: "%env(REDIS_URL_VALIDTOKEN)%"
```

If I rollback the upgrade, it starts working again.

**Comment by derrabus** [comment]
Please provide a [reproducer](https://symfony.com/doc/current/contributing/code/reproducer.html) for your bug.

**Comment by HypeMC** [comment]
Seems to be related to https://github.com/symfony/symfony/pull/50718

**Comment by aboks** [comment]
Small reproduction script for this issue:

```php
<?php
// Install symfony/dependency-injection (6.3 works ok, 6.4 fails)

use Symfony\Component\DependencyInjection\ContainerBuilder;

require_once __DIR__ . "/vendor/autoload.php";

class MyClass {
	public function __construct(\DateTimeImmutable $now_datetime) {}
}

$container = new ContainerBuilder();
$container->autowire(\DateTimeImmutable::class . ' $now_datetime', \DateTimeImmutable::class);
$container->autowire(MyClass::class, MyClass::class)->setPublic(true);
$container->compile();

$container->get(MyClass::class);

print "OK";
```
The essence is that the argument name (here: `$now_datetime`) contains an underscore. A name without an underscore works fine. Possibly there are other characters that introduce the same problem.

**Comment by derrabus** [comment]
Thanks. I've run `git bisect` on that reproducer and found 8d60a7ec0d6f6c2d0038123d653a0927b291a0a9 as the first commit on which it fails. That's indeed #50718 by @nicolas-grekas.

**Comment by paullallier** [comment]
Thank you for digging / the reproducer - I suspect it would have taken me quite a while...

**Comment by nicolas-grekas** [comment]
This should be fixed by #52836.
Please confirm if you can.

**Comment by paullallier** [comment]
Works for me - all my tests pass. Thanks!
